pkg/plugin/config_test.go

```diff
@@ -0,0 +1,148 @@
+		name       string
+		args       []string
+		envs       map[string]string
+		wantConfig *PluginConfig
+	}{
+		{
+			name: "all_envs_specified",
+			envs: map[string]string{
+				"GITHUB_APP_ID":              testGitHubAppID,
+				"GITHUB_APP_INSTALLATION_ID": testGitHubAppInstallationID,
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+			wantConfig: &PluginConfig{
+				GitHubAppID:             testGitHubAppID,
+				GitHubAppInstallationID: testGitHubAppInstallationID,
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+		},
+		{
+			name: "all_flags_specified",
+			args: []string{
+				"-github-app-id", testGitHubAppID,
+				"-github-app-installation-id", testGitHubAppInstallationID,
+				"-github-app-private-key-pem", testPrivateKeyString,
+			},
+			wantConfig: &PluginConfig{
+				GitHubAppID:             testGitHubAppID,
+				GitHubAppInstallationID: testGitHubAppInstallationID,
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			gotConfig := &PluginConfig{}
+			set := cli.NewFlagSet(cli.WithLookupEnv(cli.MapLookuper(tc.envs)))
+			set = gotConfig.ToFlags(set)
+			if err := set.Parse(tc.args); err != nil {
+				t.Errorf("unexpected flag set parse error: %v", err)
+			}
+			if diff := cmp.Diff(tc.wantConfig, gotConfig); diff != "" {
+				t.Errorf("Config unexpected diff (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestPluginConfig_Validate(t *testing.T) {
+	t.Parallel()
+
+	testPrivateKeyString, _ := testGeneratePrivateKey(t)
+
+	cases := []struct {
+		name    string
+		cfg     *PluginConfig
+		wantErr string
+	}{
+		{
+			name: "success",
+			cfg: &PluginConfig{
+				GitHubAppID:             testGitHubAppID,
+				GitHubAppInstallationID: testGitHubAppInstallationID,
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+		},
+		{
+			name: "empty_github_app_id",
+			cfg: &PluginConfig{
+				GitHubAppID:             "",
+				GitHubAppInstallationID: testGitHubAppInstallationID,
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			wantErr: "GITHUB_APP_ID is empty",
+		},
+		{
+			name: "empty_github_app_installation_id",
+			cfg: &PluginConfig{
+				GitHubAppID:             testGitHubAppID,
+				GitHubAppInstallationID: "",
+				GitHubAppPrivateKeyPEM:  testPrivateKeyString,
+			},
+			wantErr: "GITHUB_APP_INSTALLATION_ID is empty",
+		},
+		{
+			name: "empty_github_app_private_key_pem",
+			cfg: &PluginConfig{
+				GitHubAppID:             testGitHubAppID,
+				GitHubAppInstallationID: testGitHubAppInstallationID,
+				GitHubAppPrivateKeyPEM:  "",
+			},
+			wantErr: "GITHUB_APP_PRIVATE_KEY_PEM is empty",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			err := tc.cfg.Validate()
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
+				t.Errorf("Unexpected err: %s", diff)
```

Comment:
I don't think you need the prefix message here:

```suggestion
				t.Error(diff)
```

Reply:
Updated.